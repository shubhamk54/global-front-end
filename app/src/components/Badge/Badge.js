import PropTypes from 'prop-types';
import classNames from 'classnames';
import React from 'react';
import './Badge.scss';


function Badge({ ...props }) {
  const { displayName, variant } = props;
  const badgeClasses = classNames(
    'badge',
    variant && `badge-${variant}`,
  );

  return <span className={displayName && badgeClasses}>{displayName}</span>;
}

Badge.defaultProps = {
  variant: 'info',
};

Badge.propTypes = {
  variant: PropTypes.oneOf([
    'danger',
    'success',
    'info',
  ]).isRequired,
  displayName: PropTypes.string.isRequired,
};


export default Badge;
