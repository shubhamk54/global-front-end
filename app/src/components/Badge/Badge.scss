.badge {
  display: inline-block;
  padding: 0.25em 0.4em;
  text-align: center;
  border-radius: 0.25rem;
  color: #fff;

  &-success {
    background-color: #28a745;
  }

  &-danger {
    background-color: #c72323;
  }
}
