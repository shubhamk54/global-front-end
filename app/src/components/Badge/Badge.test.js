import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Badge from './Badge.js';

const fakeProps = { displayName: 'Active' };

describe('Snapshot testing for Badge component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Badge {...fakeProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Unit tests for Badge component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Badge {...fakeProps} />,
    );
  });

  it('Badge renders the text inside it', () => {
    const span = wrapper.find('.badge');
    expect(span.text()).toBe(fakeProps.displayName);
  });

  it('Badge renders the text inside it', () => {
    const span = wrapper.find('.badge');
    expect(span.text()).not.toBe('Invalid');
  });
});
