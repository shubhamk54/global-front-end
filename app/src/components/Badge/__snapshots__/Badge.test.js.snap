// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Badge component matches the snapshot 1`] = `
<span
  className="badge badge-info"
>
  Active
</span>
`;
